File: src/main/java/org/retrolauncher/app/games/infrastructure/cli/commands/StartGameCommand.java
package org.retrolauncher.app.games.infrastructure.cli.commands;

import org.retrolauncher.app.games.application.usecases.StartGameUseCase;
import picocli.CommandLine;

import java.util.concurrent.Callable;

@CommandLine.Command(
        name = "game:start",
        version = "1.0",
        description = "Inicia jogo dado um id"
)
public class StartGameCommand implements Callable<Integer> {
    private final StartGameUseCase startGameUseCase;

    @CommandLine.Parameters(index = "0", description = "O id do jogo")
    private String id;

    public StartGameCommand(StartGameUseCase startGameUseCase) {
        this.startGameUseCase = startGameUseCase;
    }

    @Override
    public Integer call() {
        this.startGameUseCase.execute(this.id);
        return 0;
    }
}

File: backend/src/app/games/internal/delivery/cli/start_game_command.go
package cli

import "retrolauncher/backend/src/app/games/internal/application"

type StartGameCommand interface {
	Execute(id string)
}

type startGameCommand struct {
	startGame application.StartGame
}

func NewStartGameCommand(startGame application.StartGame) StartGameCommand {
	return &startGameCommand{
		startGame: startGame,
	}
}

func (s *startGameCommand) Execute(id string) {
	s.startGame.Execute(application.StartGameInput{
		GameId: id,
	})
}
